src/merge_bed.py

```diff
@@ -30,8 +31,48 @@ def read_bed_file(f: TextIO) -> list[BedLine]:
 
 
 def merge(f1: list[BedLine], f2: list[BedLine], outfile: TextIO) -> None:
-    """Merge features and write them to outfile."""
-    # FIXME: I have work to do here!
+    """Merge features and write them to outfile.
+    
+    """
+    # Input bedlines er sorted. 
+    # Hver region kun én nucleotid lang. 
+
+    a=0
+    b=0
+
+    while a <= len(f1)-1 and b <= len(f2)-1:
+        if f1[a][0]==f2[b][0]:
+            if f1[a][1]<f2[b][1]:
+                print_line(f1[a],outfile)
+                a+=1
+            elif f1[a][0]>f2[b][0]:
+                print_line(f1[b],outfile)
+                b+=1
+            elif f1[a][1]==f2[b][1]:
+                print_line(f1[a],outfile)
+                a+=1
+                print_line(f1[b],outfile)
+                b+=1
+                
+            
+        else:
+            if f1[a][0]<f2[b][0]:
+                print_line(f1[a],outfile)
+                a+=1
+            elif f1[a][0]>f2[b][0]:
+                print_line(f1[b],outfile)
+                b+=1
+    
+    if a != len(f1)-1:
+        for i in range(a,len(f1)-1):
+            print_line(f1[i],outfile)
+    elif b != len(f2)-1:
+        for i in range(b,len(f2)-1):
+            print_line(f1[i],outfile)
```

Comment:
There is nothing wrong with iterating through an empty list--the loop won't run, so you don't do anything. If you don't mind doing that, then this code gets a lot simpler.